File: app/src/main/java/com/zane/smapiinstaller/logic/ActivityResultHandler.java
package com.zane.smapiinstaller.logic;

import android.content.Intent;

import java.util.concurrent.ConcurrentHashMap;

import java.util.function.BiConsumer;

/**
 * @author Zane
 */
public class ActivityResultHandler {
    public static final int REQUEST_CODE_APP_INSTALL = 1001;
    public static final int REQUEST_CODE_ALL_FILES_ACCESS_PERMISSION = 1002;
    public static final int REQUEST_CODE_OBB_FILES_ACCESS_PERMISSION = 1003;

    public static ConcurrentHashMap<Integer, BiConsumer<Integer, Intent>> listenerMap = new ConcurrentHashMap<>();

    public static void registerListener(int requestCode, BiConsumer<Integer, Intent> listener) {
        listenerMap.put(requestCode, listener);
    }

    public static void triggerListener(int requestCode, int resultCode, Intent data) {
        BiConsumer<Integer, Intent> biConsumer = listenerMap.get(requestCode);
        if(biConsumer != null) {
            biConsumer.accept(resultCode, data);
        }
    }
}

File: app/src/main/java/com/zane/smapiinstaller/logic/ActivityResultHandler.kt
package com.zane.smapiinstaller.logic

import android.content.Intent
import java.util.concurrent.ConcurrentHashMap

/**
 * @author Zane
 */
object ActivityResultHandler {
    const val REQUEST_CODE_APP_INSTALL = 1001
    const val REQUEST_CODE_ALL_FILES_ACCESS_PERMISSION = 1002
    const val REQUEST_CODE_OBB_FILES_ACCESS_PERMISSION = 1003
    const val REQUEST_CODE_DATA_FILES_ACCESS_PERMISSION = 1004
    var listenerMap = ConcurrentHashMap<Int, (requestCode: Int, data: Intent?) -> Unit>()

    @JvmStatic
    fun registerListener(requestCode: Int, listener: (requestCode: Int, data: Intent?) -> Unit) {
        listenerMap[requestCode] = listener
    }

    fun triggerListener(requestCode: Int, resultCode: Int, data: Intent?) {
        val biConsumer = listenerMap[requestCode]
        biConsumer?.invoke(resultCode, data)
    }
}